lib/resizer.js

```diff
@@ -40,85 +43,118 @@ exports.work = (imgEvent) => {
 }
 
 // upload to final s3 destination
-exports.upload = (fileStream, filePath, fileName, variation) => {
-  if (variation) {
-    let parts = path.parse(fileName);
-    fileName = `${parts.name}_${variation}${parts.ext}`;
-  }
+exports.upload = (tmpFile, destPath, contentType) => {
+  let fileName = path.basename(tmpFile);
   return Q.ninvoke(s3, 'upload', {
-    Body: fileStream,
+    Body: fs.createReadStream(tmpFile),
+    Bucket: process.env.DESTINATION_BUCKET,
+    Key: `${destPath}/${fileName}`,
+    ContentType: contentType
+  });
+}
+
+// copy between s3 locations
+exports.copy = (tmpBucket, tmpKey, destPath, contentType) => {
+  let fileName = path.basename(tmpKey);
+  return Q.ninvoke(s3, 'copyObject', {
+    CopySource: `${tmpBucket}/${tmpKey}`,
     Bucket: process.env.DESTINATION_BUCKET,
-    Key: `${filePath}/${fileName}`
+    Key: `${destPath}/${fileName}`,
+    MetadataDirective: 'REPLACE',
```

Comment:
Have to REPLACE the metadata to get content-type to stick.  (We don't want the source's metadata anyways).